### ISSUE TITLE
Prometheus metrics endpoint returns 500 instead of metrics

### ISSUE THREAD

**Issue by janLo** [coder/coder#11451]
Our instances stopped to respond to metric requests:
```
❯ curl http://coder-service2.xxx:2112/
An error has occurred while serving metrics:

132 error(s) occurred:
* collected metric "agent_sessions_errors_total" { label:{name:"agent_name"  value:"main"}  label:{name:"error_type"  value:"output_io_copy"}  label:{name:"magic_type"  value:"ssh"}  label:{name:"pty"  value:"yes"}  label:{name:"template_name"  value:"debian-like-buildenv"}  label:{name:"username"  value:"xxx"}  label:{name:"workspace_name"  value:"muts"}  counter:{value:11}} was collected before with the same name and label values
* collected metric "agent_sessions_total" { label:{name:"agent_name"  value:"main"}  label:{name:"magic_type"  value:"ssh"}  label:{name:"pty"  value:"no"}  label:{name:"template_name"  value:"debian-like-buildenv"}  label:{name:"username"  value:"xxx"}  label:{name:"workspace_name"  value:"muts"}  counter:{value:6}} was collected before with the same name and label values
* collected metric "agent_sessions_total" { label:{name:"agent_name"  value:"main"}  label:{name:"magic_type"  value:"ssh"}  label:{name:"pty"  value:"yes"}  label:{name:"template_name"  value:"debian-like-buildenv"}  label:{name:"username"  value:"xxx"}  label:{name:"workspace_name"  value:"muts"}  counter:{value:18}} was collected before with the same name and label values
[...]
```

It seem to have happened with the upgrade to 2.6.0.

**Comment by johnstcn** [comment]
@janLo do you have `CODER_PROMETHEUS_COLLECT_AGENT_STATS=true`? I think you should be able to set this to `false` while we investigate the root cause.

**Comment by janLo** [comment]
Thank you, I'll try that on monday. One thing to note: I updated from 2.5.0.

**Comment by Emyrk** [comment]
@spikecurtis are you working on this, or do you want me to take this up?

**Comment by janLo** [comment]
Thank you, looking forward to the release!
:heart:

**Comment by johnstcn** [comment]
released in 2.7.0 https://github.com/coder/coder/releases/tag/v2.7.0
